flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/serializer/jsondebezium/JsonDebeziumSchemaChangeImplV2.java

```diff
@@ -337,6 +337,9 @@ public String buildDorisTypeName(JsonNode column) {
         String dorisTypeName;
         switch (sourceConnector) {
             case MYSQL:
+                if (sourceTypeName.equals("DATETIME") || sourceTypeName.equals("TIMESTAMP")) {
+                    length = 20 + length;
+                }
                 dorisTypeName = MysqlType.toDorisType(sourceTypeName, length, scale);
```

Comment:
Can it be added directly to this method?

Reply:
I have added it to MysqlType.toDorisType()